one-ui/src/main/java/com/marklogic/hub/oneui/controllers/EnvironmentController.java

```diff
@@ -112,8 +113,11 @@ public void onError(String commandName, Exception exception) {
         String directory = payload.get("directory").asText("");
         // setting the project directory will resolve any relative paths
         try {
+            Path directoryPath = Paths.get(directory);
             if (StringUtils.isEmpty(directory)) {
                 throw new BadRequestException("Missing required directory property");
+            } else if (!(directoryPath.isAbsolute() || directoryPath.toFile().exists())) {
+                throw new IOException("Invalid directory supplied");
```

Comment:
I recommend including the directory path in here for more context - e.g. 

    throw new IllegalArgumentException("Invalid project directory: " + directory);

I also try to avoid IOException whenever possible, as it's a checked exception. For almost all IO errors, it's unlikely that a client can do anything meaningful with it other than rethrow it.